sys/stat_unsupported.go

```diff
@@ -0,0 +1,11 @@
+//go:build (!((amd64 || arm64 || riscv64) && linux) && !((amd64 || arm64) && (darwin || freebsd)) && !((amd64 || arm64) && windows)) || js
```

Comment:
Wondering if we need to do this by CPU when it's a OS thing, but it's likely convention between "supported" (we test it) vs. "unsupported" (we don't).

Reply:
I agree it is mostly about what we are testing, as there would be some stat_t which would be compatible with our conversion logic.

I think this was also trying to say "64-bit" since the stat lengths are different on 32. I'll backfill a comment which I suppose we could support later (even if not test it) if someone needed precise stat on a 32-bit platform.